Keyclock customization (#113)
vshn/appcat-user-docs: docs/modules/ROOT/pages/vshn-managed/keycloak/customization.adoc

> = Customization

Reviewer: Shouldn't we be more specific about what customizations we're talking about? Is it about custom configuration? Or a custom Keycloak image?

Reviewer: ```suggestion
= Custom Themes and Providers
```

Reviewer: BTW: Why is it needed to have custom themes this way? I thought we have a custom extension by Inventage which makes it possible to upload themes without having to have a custom image? https://products.vshn.ch/appcat/keycloak.html#_configurable_theme If this isn't the case, do we have to update the product description?

Author: > Shouldn't we be more specific about what customizations we're talking about? Is it about custom configuration? Or a custom Keycloak image?

I expect that we will have more custom things for Keycloak in the future thus the reason of the main vague title. As a subtitle I aded `Themes and providers`

Reviewer 2: They have a theme editor as well. However, the capabilities are very limited. You can define some colors and an image (that is accessible somewhere on the internet) that you can specify for the welcome theme.
There is currently no option to specify more detailed themes. Also it's not possible to use this to adjust the look and feel when you are logged in (only the welcome page).
We therefore also support providing native keycloak themes.

> This page will describe how you can customize your Keycloak instance.

Reviewer: Same here, be more specific about what customizations we're talking about

Author: The specifics should be in the subtitles. For now there is only one customisation but we might have more in the near future.

Reviewer: OK, let's see how that turns out

== Themes and providers

`VSHNKeycloak` allows to use custom providers and themes.

> xref:vshn-managed/how-tos/custom-image-keycloak.adoc[Create] a custom Docker image and use it to customize your Keycloak instance as described below.

Reviewer: ```suggestion
xref:vshn-managed/how-tos/custom-image-keycloak.adoc[Create] a custom container image and use it to customize your Keycloak instance as described below.
```

Just a nit, but I prefer to use the word "container" instead of Docker, as Docker is the name of the product...

> .Example to create a Keycloak instance with custom image

Reviewer: ```suggestion
.Example to create a Keycloak instance an image containing custom themes and providers
```

[source,yaml]
----
apiVersion: vshn.appcat.vshn.io/v1
kind: VSHNKeycloak
metadata:
  name: keycloak-app1-prod
  namespace: prod-app
spec:
  parameters:
    service:
      version: "23"

>       customizationImage: https://registry/user/image:tag # <1>

Reviewer 2: This is wrong. The `customizationImage` is a type that has the subkeys  `image` for the actual image as well as `imagePullSecretRef` for the optional pullSecret (if that is required)

Author: oh thanks for the note, I will adjust the code

    size:
      plan: standard-2
  writeConnectionSecretToRef:
    name: keycloak-creds
----

> <1> Your docker image url.

Reviewer: ```suggestion
<1> Container image url.
```

= Customization

This page will describe how you can customize your Keyclock instance.

== Themes and providers

> To customize providers and themes customization images parameter should be used.

Reviewer: I don't understand this sentence

Author: I rewrote the whole paragraph.

> Include your own image with custom themes and providers. Place your themes in the `/themes` directory and your providers in the `/providers` directory within your custom Docker image.

Reviewer 3: I feel like there's some context missing here for people who don't know how this is supposed to work. Maybe go into more detail. Like an introduction that a custom image is needed to add the customizations or how the customization works.

Because right now it's not really clear what to do, if you've never used this system for customization.

Reviewer: I'm with Simon, I have no idea what we're talking about here. Put yourself into the position of an end-user, not having any idea about how our system works, with that hat on, it lacks a lot of context.

= Customization

This page will describe how you can customize your Keycloak instance.

== Themes and providers

> To customize providers and themes customization images parameter should be used.

Reviewer 3: ```suggestion
To customize providers and themes `customizationImage` parameter should be used.
```

Author: I changed the wording as it was bad :)
